### PR TITLE
Replace "primary" and "default" builds with "f407-discovery"

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**
It has been asked whether the primary build serves any purpose, and there hasn't been any answer. Therefore we shall yeet with extreme prejudice.

- Remove "rusefi_bundle"
- Remove "rusefi_bundle_default"
- Introduce "rusefi_bundle_f407-discovery"
- Reset one more file after board-specific gen-config
- Get rid of a gen_config_default call of unknown purpose (maybe for #3367)
- Don't check for "all"
- Don't run gen_config_default in gen_config

This should fix:
#5016
#3182
#3085
#3367

### CODE REVIEW

**Comment by rusefillc** [comment]
let's break this PR apart

for first change can we go only incompatible rename of rusefi_bundle.zip which would break some old links but we should not care about those old links, that rename of "default" bundle should have CHANGELOG

File: .github/workflows/build-firmware.yaml [hunk 1]
```diff
@@ -335,13 +339,6 @@ jobs:
       working-directory: ./firmware/
       run: ./gen_enum_to_string.sh
 
-      # depends on gen_live_documentation.sh for instance TS_TOTAL_OUTPUT_SIZE
-    - name: Generate Default config
-      if: ${{ env.skip != 'true' }}
```
Comment (rusefillc) [inline .github/workflows/build-firmware.yaml:340]: i have a feeling that this would break something major, let' start breaking the change apart

**Review by rusefillc — CHANGES_REQUESTED** [review]

File: misc/jenkins/functional_test_and_build_bundle/run.sh [hunk 2]
```diff
@@ -1,8 +1,6 @@
 #!/bin/bash
 
 git submodule update --init
-bash misc/jenkins/functional_test_and_build_bundle/build_current_bundle.sh
```
Comment (rusefillc) [inline misc/jenkins/functional_test_and_build_bundle/run.sh:4]: huh? just like that - just do not build for HW CI? let's extract into separate PR?

Comment (chuckwagoncomputing) [reply misc/jenkins/functional_test_and_build_bundle/run.sh:4]: HW CI doesn't run that script; I don't know what it's for.

Comment (rusefillc) [reply misc/jenkins/functional_test_and_build_bundle/run.sh:4]: Not trying to attach just want to confirm: "i am not sure what this script does but I have confidence to make interesting changes"?

Comment (rusefillc) [reply misc/jenkins/functional_test_and_build_bundle/run.sh:4]: I am fine doing https://github.com/rusefi/rusefi/commit/d3c12dc6f53c77918de227bb7cb997ce2e4faabe and https://github.com/rusefi/rusefi/commit/33a26d05f7e209f4894557fa16dc9225aae04d18

**Comment by chuckwagoncomputing** [comment]
Cherry pick as you please

**Comment by rusefillc** [comment]
@chuckwagoncomputing need your help with all this :( we do need to reduce the madness we just need to do it in smaller chunks

**Comment by chuckwagoncomputing** [comment]
OK I'll see what I can come up with.

File: .github/workflows/build-firmware.yaml (hunk 1 above)
Comment (mck1117) [reply .github/workflows/build-firmware.yaml:340]: there's always the scream test of "do it and see what screams"

**Review by mck1117 — APPROVED** [review]

File: java_console/io/src/main/java/com/rusefi/io/DfuHelper.java [hunk 3]
```diff
@@ -34,15 +34,8 @@ public static boolean sendDfuRebootCommand(JComponent parent, String signature,
...
-/*
-todo: fix https://github.com/rusefi/rusefi/issues/5016
             String signatureWithPrefix;
-            if ("all".equals(s.getBundleTarget())) {
-                signatureWithPrefix = PREFIX;
-            } else {
-                signatureWithPrefix = PREFIX + "_" + s.getBundleTarget();
-            }
-*/
+            signatureWithPrefix = PREFIX + "_" + controllerSignature.getBundleTarget();
```
Comment (rusefillc) [inline java_console/io/src/main/java/com/rusefi/io/DfuHelper.java:38]: @chuckwagoncomputing are you adding an unused variable here?

**Review by rusefillc — CHANGES_REQUESTED** [review]

File: .github/workflows/build-firmware.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@ jobs:
         # see also build-primary-bundle where default/primary bundle is build separately
         #
         build-target: [
-            all,
+            f407-discovery,
```
Comment (rusefillc) [inline .github/workflows/build-firmware.yaml]: would this add second f407-discovery?

File: java_console/io/src/main/java/com/rusefi/io/DfuHelper.java (hunk 3 above)
Comment (chuckwagoncomputing) [reply java_console/io/src/main/java/com/rusefi/io/DfuHelper.java:38]: I guess so. I just preserved it without checking to see if it was ever used.

File: .github/workflows/build-firmware.yaml (hunk 4 above)
Comment (chuckwagoncomputing) [reply .github/workflows/build-firmware.yaml]: That change isn't there anymore, I'm not sure when this review comes from.

Comment (rusefillc) [reply .github/workflows/build-firmware.yaml]: i am reviewing commits one after another. if commit history is misleading can we please squash?

**Comment by chuckwagoncomputing** [comment]
What's your current opinion of this?

**Pull request merged**